src/request_queue.js

```diff
@@ -415,15 +503,17 @@ export class RequestQueue {
 
     /**
      * Resolves to `true` if all requests were already handled and there are no more left.
-     * Due to the nature of distributed storage systems,
-     * the function might occasionally return a false negative, but it will never return a false positive.
+     * Due to the nature of distributed storage used by the queue,
+     * the function might occasionally return a false negative,
+     * but it will never return a false positive.
      *
      * @returns {Promise<Boolean>}
      */
     async isFinished() {
-        if (this.inProgressCount > 0 || this.queueHeadDict.length() > 0) return false;
+        if (this.queueHeadDict.length() > 0 || this.inProgressCount() > 0) return false;
+
         const isHeadConsistent = await this._ensureHeadIsNonEmpty(true);
-        return isHeadConsistent && this.inProgressCount === 0 && this.queueHeadDict.length() === 0;
+        return isHeadConsistent && this.queueHeadDict.length() === 0 && this.inProgressCount() === 0;
```

Comment:
This is the third time we change this in the last months. Are we sure this is finally correct or is this just a hopeful fix?

Reply:
It's just a change in order

Reply:
Ha, fair point 😆